.github/workflows/cmake.yml

```
@@ -94,16 +98,59 @@ jobs:
       - uses: actions/checkout@v4
         with:
           submodules: recursive
+      - uses: actions/checkout@v4
+        if: env.APACHE_ARROW == 'main'
+        with:
+          path: apache-arrow
+          repository: apache/arrow
       - name: Prepare ccache
         run: |
           echo "CCACHE_DIR=${PWD}/ccache" >> ${GITHUB_ENV}
       - name: Cache ccache
         uses: actions/cache@v4
         with:
           path: ccache
-          key: cmake-ubuntu-${{ matrix.cc }}-ccache-${{ hashFiles('lib/**', 'src/**', 'plugins/**', 'include/**') }}
-          restore-keys: cmake-ubuntu-${{ matrix.cc }}-ccache-
+          key: cmake-ubuntu-${{ matrix.cc }}-${{ matrix.apache-arrow }}-ccache-${{ hashFiles('lib/**', 'src/**', 'plugins/**', 'include/**', 'apache-arrow/cpp/src/**') }}
+          restore-keys: |
+            cmake-ubuntu-${{ matrix.cc }}-${{ matrix.apache-arrow }}-ccache-
+            cmake-ubuntu-${{ matrix.cc }}-ccache-
+            cmake-ubuntu-ccache-
```

```suggestion
            cmake-ubuntu-${{ matrix.cc }}--ccache-
```

Are these two keys ever set?